### PR TITLE
remove print ops before exporting onnx graph

### LINKED ISSUE #15505: [ONNX] Exporting JIT script with warning subcall fails
## 🐛 Bug

ONNX export fails when a JITted function emits a warning. I probably should fix the warning, but someone (who isn't me) who tries to export a popular open source model should probably not see
```
Unsupported prim::Constant kind: `s`. Send a bug report.
```
It looks like the issue is that the warning string gets baked into the function during JIT compilation and hence exported.

MWE:

```python
import torch

@torch.jit.script
def jit_warner(inp):
    return torch.nn.functional.sigmoid(inp) # triggers a deprecation warning

@torch.jit.script
def jit_ok(inp):
    return torch.sigmoid(inp)

class JitWarner(torch.nn.Module):
    def __init__(self):
        super().__init__()
        
    def forward(self, x):
        return jit_warner(x)
        # return jit_ok(x) # this works

torch.onnx.export(JitWarner(), torch.randn(42), '/dev/null')
```

## Environment

 - PyTorch Version (e.g., 1.0): 1.0
 - OS (e.g., Linux): Linux
 - How you installed PyTorch (`conda`, `pip`, source): `pip`
 - Python version: 3.6

## Additional context

I'm trying to export the [waveglow](https://github.com/NVIDIA/waveglow/blob/master/glow_old.py) model.

### PR DESCRIPTION
**Pull request by eellison**
Removing print ops before exporting onnx graph, fixes https://github.com/pytorch/pytorch/issues/15505

### CODE REVIEW

File: torch/csrc/jit/passes/onnx.cpp [hunk 1]
```diff
@@ -10,10 +10,28 @@
...
+    }
+    if (it->kind() == prim::Print || it->kind() == aten::warn) {
+      it.destroyCurrent();
+    }
+  }
+}
+
+void removePrintOps(std::shared_ptr<Graph>& graph) {
+  removePrintOps(graph->block());
+  EliminateDeadCode(graph);
```
Comment (driazati) [inline torch/csrc/jit/passes/onnx.cpp]: Why does this need to call DCE? That might do way more than remove print nodes

File: torch/csrc/jit/passes/onnx.cpp [hunk 2]
```diff
@@ -10,10 +10,28 @@
...
 namespace torch { namespace jit {
 
+void removePrintOps(Block * block) {
+  for (auto it = block->nodes().begin(), end = block->nodes().end(); it != end;
+       ++it) {
+    for (auto b: it->blocks()) {
+      removePrintOps(b);
+    }
+    if (it->kind() == prim::Print || it->kind() == aten::warn) {
+      it.destroyCurrent();
```
Comment (zou3519) [inline torch/csrc/jit/passes/onnx.cpp]: This is amazing, I didn't know `destroyCurrent` existed

**Review by driazati — APPROVED** [review]

**Review by apaszke — COMMENTED** [review]
We should probably print a warning that prints have been removed

**Comment by jamesr66a** [comment]
I think it would make more sense for this code to live here: https://github.com/pytorch/pytorch/blob/78d594f46cf48cff620aa3e69c40697882a503f3/torch/onnx/utils.py#L507

Is there anything preventing that?

**Pull request closed without merging**